### DISCUSSION TITLE
Getting Crash on PIC 16F877A

### DISCUSSION THREAD

**Discussion by gokulapandiyan** [lcgamboa/picsimlab#100] [Show and tell]
PICSimLab: Version "0.9.1 231210 Windows64_WX DIST"
PICSimLab: Command Line: C:\Program Files\PicsimLab\picsimlab.exe 
PICSimLab: Error open config file "C:\Users\gokul\AppData\Roaming\picsimlab/picsimlab.ini"!
PICSimLab: Using board "Arduino Uno"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_Arduino_Uno_atmega328p.hex"
PICSimLab: File not found! Creating new empty file. 
PICSimLab: Error on Port Open \\.\com6: 0x2 !
PICSimLab: Debug On=0  Type=MDB Port=1234
PICSimLab: Remote Control Port 5000
PICSimLab: Loading board demonstration code.
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_Arduino_Uno_atmega328p.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Local\Temp/picsimlab-P1kAlM/picsimlab_workspace//picsimlab.ini"
PICSimLab: Window position x=97 y=53
PICSimLab: Using board "Arduino Uno"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Local\Temp/picsimlab-P1kAlM/picsimlab_workspace//mdump_Arduino_Uno_atmega328p.hex"
PICSimLab: Error on Port Open \\.\com6: 0x2 !
PICSimLab: Debug On=0  Type=GDB Port=1234
PICSimLab: Remote Control Port 5000
Spare parts: parts[00] (Step Motor) created 
Spare parts: parts[01] (Text Box) created 
Spare parts: parts[02] (IO Virtual Term) created 
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_Arduino_Uno_atmega328p.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Roaming\picsimlab/picsimlab.ini"
PICSimLab: Window position x=513 y=218
PICSimLab: Using board "PICGenios"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC18F4520.hex"
PICSimLab: File not found! Creating new empty file. 
Error loading from file: C:\Users\gokul\AppData\Roaming\picsimlab//mdump_PICGenios_EEPROM.bin 
ERROR: Picsim->File not found!(C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC18F4520.hex)
PICSimLab: Debug On=0  Type=GDB Port=1234
PICSimLab: Remote Control Port 5000
PICSimLab: Loading board demonstration code.
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC18F4520.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Local\Temp/picsimlab-3SFf2w/picsimlab_workspace//picsimlab.ini"
PICSimLab: Window position x=0 y=0
PICSimLab: Using board "PICGenios"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Local\Temp/picsimlab-3SFf2w/picsimlab_workspace//mdump_PICGenios_PIC18F4520.hex"
Spare parts: parts[00] (Text Box) created 
Spare parts: parts[01] (IO Virtual Term) created 
PICSimLab: Debug On=0  Type=GDB Port=1234
PICSimLab: Remote Control Port 5000
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC18F4520.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Roaming\picsimlab/picsimlab.ini"
PICSimLab: Window position x=72 y=7
PICSimLab: Using board "PICGenios"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: File not found! Creating new empty file. 
ERROR: Picsim->File not found!(C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex)
Spare parts: parts[00] (Text Box) created 
Spare parts: parts[01] (IO Virtual Term) created 
PICSimLab: Debug On=0  Type=GDB Port=1234
PICSimLab: Remote Control Port 5000
Error loading from file: C:\Users\gokul\MPLABXProjects\simulation.X\dist\default\production\/mdump_PICGenios_EEPROM.bin 
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Roaming\picsimlab/picsimlab.ini"
PICSimLab: Window position x=-8 y=-8
PICSimLab: Using board "PICGenios"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Roaming\picsimlab/picsimlab.ini"
PICSimLab: Window position x=-8 y=-8
PICSimLab: Using board "PICGenios"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: Debug On=0  Type=GDB Port=1234
PICSimLab: Remote Control Port 5000
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Roaming\picsimlab/picsimlab.ini"
PICSimLab: Window position x=-8 y=-8
PICSimLab: Using board "PICGenios"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: Debug On=1  Type=GDB Port=1234
PICSimLab: Remote Control Port 5000
PICSimLab: Debug On=1  Type=GDB Port=1234
PICSimLab: Remote Control Port 5000
Error loading from file: C:\Users\gokul\MPLABXProjects\simulation.X\dist\default\production\/mdump_PICGenios_EEPROM.bin 
Error loading from file: C:\Users\gokul\MPLABXProjects\simulation.X\dist\default\production\/mdump_PICGenios_EEPROM.bin 
Error loading from file: C:\Users\gokul\MPLABXProjects\simulation.X\dist\default\production\/mdump_PICGenios_EEPROM.bin 
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Roaming\picsimlab/picsimlab.ini"
PICSimLab: Window position x=-8 y=-8
PICSimLab: Using board "PICGenios"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: Debug On=0  Type=GDB Port=1234
PICSimLab: Remote Control Port 5000
Error loading from file: C:\Users\gokul\MPLABXProjects\simulation.X\dist\default\production\/mdump_PICGenios_EEPROM.bin 
Error loading from file: C:\Users\gokul\MPLABXProjects\simulation.X\dist\default\production\/mdump_PICGenios_EEPROM.bin 
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Roaming\picsimlab/picsimlab.ini"
PICSimLab: Window position x=258 y=49
PICSimLab: Using board "PICGenios"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: Debug On=0  Type=GDB Port=1234
PICSimLab: Remote Control Port 5000
Error loading from file: C:\Users\gokul\MPLABXProjects\simulation.X\dist\default\production\/mdump_PICGenios_EEPROM.bin 
Error loading from file: C:\Users\gokul\MPLABXProjects\simulation.X\dist\default\production\/mdump_PICGenios_EEPROM.bin 
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Roaming\picsimlab/picsimlab.ini"
PICSimLab: Window position x=204 y=21
PICSimLab: Using board "PICGenios"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: Debug On=0  Type=GDB Port=1234
PICSimLab: Remote Control Port 5000
Error loading from file: C:\Users\gokul\MPLABXProjects\simulation.X\dist\default\production\/mdump_PICGenios_EEPROM.bin 
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Roaming\picsimlab/picsimlab.ini"
PICSimLab: Window position x=204 y=0
PICSimLab: Using board "PICGenios"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: Saving "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex"
PICSimLab: End Board Simulation.
PICSimLab: Load Config from file "C:\Users\gokul\AppData\Roaming\picsimlab/picsimlab.ini"
PICSimLab: Window position x=204 y=0
PICSimLab: Using board "PICGenios"
PICSimLab: Instance number 0
PICSimLab: Opening "C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGeniosPICSimLab Error: EXCEPTION_ACCESS_VIOLATION

**Comment by lcgamboa** [comment]
Hi @gokulapandiyan ,

after this crash what happened? Can you open the PICSimLab? Can you select the PICGenios board?  The file C:\Users\gokul\AppData\Roaming\picsimlab/mdump_PICGenios_PIC16F877A.hex may be corrupted (try to delete it). 
I need more information to reproduce the error.

**Reply by gokulapandiyan** [reply]
@lcgamboa I resolved the issue which is raised by the corrupted .hex
Thanks for considering  :)
